addons/mail/static/src/js/thread.js

```diff
@@ -110,6 +120,12 @@ var Thread = Widget.extend({
             ORDER: ORDER,
         }));
 
+        var attachments = _.flatten(_.map(messages, function(msg){
+            return msg.attachment_ids
+        }));
```

Comment:
The field `attachment_ids` on `mail.message` is a Many2Many, so an attachment may be linked to several messages (at least, the models allow it). So i think that you should removed duplicates (using `_.uniq`).